docs/markdown/Docker/docker.md

```diff
@@ -61,31 +61,39 @@ docker_image(
 
 Adding dependencies to your `docker_image` targets
 --------------------------------------------------
 
-A Dockerfile executes in a _context_ - a set of files that the commands in the Dockerfile can reference, e.g., by copying them into the image).
+A Dockerfile is built in a _context_ - a set of files that the commands in the Dockerfile can reference, e.g., by copying them into the image.
 
-When you run `docker` directly, the context is usually a directory within your repo. That directory must contain the Dockerfile (typically at the root of the context) and any files that the build requires. If those files are themselves the product of a build step, or if they are sources from elsewhere in the repo, then you have to copy them into the context.
+When you run `docker build` directly, the context is usually a directory within your repo containing the Dockerfile (typically at the root of the context) and any files that the build requires. If those files were themselves the product of a build step, or if they were sources from elsewhere in the repo, then you would have to copy them into the context.
 
-Pants, however, takes care of assembling the context for you. It does so using the dependencies of the `docker_image` target. 
+Pants, however, takes care of assembling the context for you. It does so using the dependencies of the [`docker_image`](doc:reference-docker_image) target, which can include:
 
-A [`docker_image`](doc:reference-docker_image) can depend on loose files belonging to [`file` / `files` targets](doc:resources#files), and on artifacts packaged from a variety of targets, such as [`pex_binary`](doc:reference-pex_binary) , [`python_distribution`](doc:reference-python_distribution), [`archive`](doc:reference-archive), or any other target that can be built via the [package](doc:reference-package) goal.
+- Loose files specified using  [`file` / `files` targets](doc:resources#files).
+- Artifacts packaged from a variety of targets, such as [`pex_binary`](doc:reference-pex_binary) , [`python_distribution`](doc:reference-python_distribution), [`archive`](doc:reference-archive), and any other target that can be built via the [package](doc:reference-package) goal, including other docker images.
 
 The context is assembled as follows:
 
 - The sources of `file` / `files` targets are assembled at their relative path from the repo root.
-- The artifacts of any packageable targets are built, as if by running `./pants package`, and placed in the context using the artifact's `output_path` field.
+- The artifacts of any packaged targets are built, as if by running `./pants package`, and placed in the context using the artifact's `output_path` field.
   - The `output_path` defaults to the scheme `path.to.directory/tgt_name.ext`, e.g. `src.python.helloworld/bin.pex`.
 
-### Dependency inference for `pex_binary`
+### Dependency inference support
```

Comment:
Not sure what to do with this section - trying to make it clear that it's not just pex_binary that is supported - but not sure we need to go to down the path of spelling out every single supported inference either.